src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs

```diff
@@ -77,31 +71,59 @@ private static bool IsNetCoreFramework(string runSettingsXml)
         /// <returns>true if runSettings was modified; false otherwise.</returns>
         private static bool TryAddFakesDataCollectorSettings(
             XmlDocument runSettings,
-            IEnumerable<string> sources)
+            IEnumerable<string> sources,
+            bool isNetFramework)
         {
             // If user provided fakes settings don't do anything
             if (XmlRunSettingsUtilities.ContainsDataCollector(runSettings.CreateNavigator(), FakesMetadata.DataCollectorUri))
             {
                 return false;
             }
 
-            Func<IEnumerable<string>, string> configurator;
+            // A new Fakes Congigurator API makes the decision to add the right datacollector uri to the configuration
+            // There now exist two data collector URIs to support two different scenarios. The new scanrio involves 
+            // using the CLRIE profiler, and the old involves using the Intellitrace profiler (which isn't supported in 
+            // .NET Core scenarios). The old API still exists for fallback measures. 
 
-            // fakes supported?
-            if (!TryGetFakesDataCollectorConfigurator(out configurator))
+            var newConfigurator = TryGetFakesNewDataCollectorConfigurator();
+            if (newConfigurator != null)
+            {
+                var fakesSettings = newConfigurator(sources, isNetFramework);
+                XmlRunSettingsUtilities.InsertDataCollectorsNode(runSettings.CreateNavigator(), fakesSettings);
+                return true;
+            }
+
+            return AddFallbackFakesSettings(runSettings, sources, isNetFramework);
+        }
+
+        private static bool AddFallbackFakesSettings(
+            XmlDocument runSettings,
+            IEnumerable<string> sources,
+            bool isNetFramework)
+        {
+
+            // The fallback settings is for the old implementation of fakes 
+            // that only supports .Net Framework versions
+            if (!isNetFramework)
+            {
+                return false;
+            }
+
+            Func<IEnumerable<string>, string> oldConfigurator = TryGetFakesDataCollectorConfigurator();
+            if (oldConfigurator == null)
             {
                 return false;
             }
 
             // if no fakes, return settings unchanged
-            var fakesConfiguration = configurator(sources);
+            var fakesConfiguration = oldConfigurator(sources);
             if (fakesConfiguration == null)
             {
                 return false;
             }
 
             // integrate fakes settings in configuration
-            // if the settings don't have any data collector settings, populate with empty settings
+            // if the settings doesn't have any data collector settings, populate with empty settings
```

Comment:
I think you were right when you used don't

Reply:
This has been updated hence too